csds-client/client/util/util.go

```diff
@@ -392,3 +346,98 @@ func PrintDetailedConfig(response proto.Message, opts ClientOptions) error {
 	}
 	return nil
 }
+
+// ConnWithJwt connects to uri with jwt authentication
+func ConnWithJwt(opts client.ClientOptions) (*grpc.ClientConn, error) {
+	if opts.Jwt == "" {
+		return nil, errors.New("missing jwt file")
+	}
+	switch opts.Platform {
+	case "gcp":
+		scope := "https://www.googleapis.com/auth/cloud-platform"
+		pool, err := x509.SystemCertPool()
+		if err != nil {
+			return nil, err
+		}
+		creds := credentials.NewClientTLSFromCert(pool, "")
+		perRPC, err := oauth.NewServiceAccountFromFile(opts.Jwt, scope)
+		if err != nil {
+			return nil, err
+		}
+
+		clientConn, err := grpc.Dial(opts.Uri, grpc.WithTransportCredentials(creds), grpc.WithPerRPCCredentials(perRPC))
+		if err != nil {
+			return nil, err
+		}
+		return clientConn, nil
+	default:
+		return nil, fmt.Errorf("%s platform is not supported, list of supported platforms: gcp", opts.Platform)
+	}
+}
+
+// ConnWithAutoGcp connects to uri on gcp with auto authentication
+func ConnWithAutoGcp(opts client.ClientOptions) (*grpc.ClientConn, error) {
```

Comment:
Same comment as in previous case, let's make interface more specific by passing concrete parameters.